doc/move_group_interface/src/move_group_interface_tutorial.cpp

```diff
@@ -1,394 +1,648 @@
+    move_group.setMaxVelocityScalingFactor(1);
+
+    // Adding or Removing Objects
+    // ^^^^^^^^^^^^^^^^^^^^^^^^^^
+    //
+    // First Define a collision object ROS message.
+    moveit_msgs::CollisionObject collision_object;
+    collision_object.header.frame_id = move_group.getPlanningFrame();
+
+    // Then set the id of the object is used to identify it.
+    collision_object.id = "box1";
+
+    // Now we can Define a box to add to the world.
+    shape_msgs::SolidPrimitive primitive;
+    primitive.type = primitive.BOX;
+    primitive.dimensions.resize(3);
+    primitive.dimensions[0] = 0.02;
+    primitive.dimensions[1] = 0.4;
+    primitive.dimensions[2] = 0.4;
+
+    // Define a pose for the box. (specified relative to frame_id)
+    geometry_msgs::Pose box_pose;
+    box_pose.orientation.w = 1.0;
+    box_pose.position.x = 0.3;
+    box_pose.position.y = -0.25;
+    box_pose.position.z = 0.5;
+
+    collision_object.primitives.push_back(primitive);
+    collision_object.primitive_poses.push_back(box_pose);
+    collision_object.operation = collision_object.ADD;
+
+    collision_objects.push_back(collision_object);
+    object_ids.push_back(collision_object.id);
+
+    // Now, let's add the collision object into the world.
+    planning_scene_interface.addCollisionObjects(collision_objects);
+    // END_SUB_TUTORIAL
+
+    // RViz
+    ROS_INFO_NAMED("tutorial", "Add an object into the world");
+    visual_tools.deleteAllMarkers();
+    visual_tools.publishText(text_pose, "Add Object", rvt::WHITE, rvt::XLARGE);
+    visual_tools.trigger();
+    visual_tools.prompt("Press 'next' once the object appears to plan and move around the obstacle");
+  }
+
+  void plan_obstacle()
+  {
+    // Set the target
+    target_pose.orientation.w = 0.707;
+    target_pose.orientation.x = 0.000;
+    target_pose.orientation.y = 0.000;
+    target_pose.orientation.z = 0.707;
+    target_pose.position.x = 0.5;
+    target_pose.position.y = -0.2;
+    target_pose.position.z = 0.5;
+    move_group.setPoseTarget(target_pose);
+
+    // Plan
+    bool success = (move_group.plan(my_plan) == moveit::planning_interface::MoveItErrorCode::SUCCESS);
+    ROS_INFO_NAMED("tutorial", "Visualizing plan 6 (pose goal move around cuboid) %s", success ? "" : "FAILED");
+
+    // Visualize the plan in RViz'
+    visual_tools.deleteAllMarkers();
+    visual_tools.publishText(text_pose, "Pose Goal Avoiding Obstacle", rvt::WHITE, rvt::XLARGE);
+    visual_tools.publishAxisLabeled(target_pose, "point");
+    visual_tools.publishTrajectoryLine(my_plan.trajectory_, joint_model_group);
+    visual_tools.trigger();
+  }
+
+  void remove_obstacle()
+  {
+    ROS_INFO_NAMED("tutorial", "Remove the object from the world");
+
+    // BEGIN_SUB_TUTORIAL remove_obj
+    //
+    // Now, let's remove the collision object from the world.
+    planning_scene_interface.removeCollisionObjects(object_ids);
+    object_ids.pop_back();
```

Comment:
This only removes the last value in the array but the above method removes the entire array.